Services/Component/classes/Setup/class.ilComponentUpdatePluginObjective.php

```diff
@@ -242,6 +243,7 @@ public function raise($a_component, $a_event, $a_parameter = ""): void
         };
         $GLOBALS["DIC"]["ilObjDataCache"] = new ilObjectDataCache();
         $GLOBALS["DIC"]["ilSetting"] = new ilSetting();
+        $GLOBALS["DIC"]["component.factory"] = $component_factory;
```

Comment:
@mjansenDatabay Would you be so kind and add that `component.repository` here as well?

Reply:
On it =)

Reply:
Found some more oddities here...